### PR TITLE
feat(Tooltip): expose data input to be passed as template outlet context when rendering the tooltip

### PR DESCRIPTION
**Pull request by Licen-it**
Closes carbon-design-system/carbon-components-angular#2853

Reintroduce `data` input to be passed as `ngTemplateOutletContext` when `description` is a template

#### Changelog

**New**

* added `data` input to Tooltip and TooltipDesctiption

**Changed**

* fixed IconButton stories that were showing incorrect input label (`cdsButton` instead of `kind`)

### CODE REVIEW

File: src/tooltip/definition-tooptip.component.ts [hunk 1]
```diff
@@ -57,6 +57,10 @@ export class TooltipDefinition extends PopoverContainer {
 	 * The string or template content to be exposed by the tooltip.
 	 */
 	@Input() description: string | TemplateRef<any>;
+	/**
+	 * Optional data for templates
+	 */
+	@Input() data = {}
```
Comment (Akshat55) [inline src/tooltip/definition-tooptip.component.ts:63]: Let's use `context` or perhaps `templateContext` to make it more descriptive.

```suggestion
	/**
	 * Optional data for templates
	 */
	@Input() context: any;
```

File: src/tooltip/tooltip.component.ts [hunk 2]
```diff
@@ -69,6 +69,10 @@ export class Tooltip extends PopoverContainer implements AfterContentChecked {
 	 * The string or template content to be exposed by the tooltip.
 	 */
 	@Input() description: string | TemplateRef<any>;
+	/**
+	 * Optional data for templates
+	 */
+	@Input() data = {}
```
Comment (Akshat55) [inline src/tooltip/tooltip.component.ts:75]: Same as above

File: src/tooltip/definition-tooptip.component.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ import { PopoverContainer } from "carbon-components-angular/popover";
 			role="tooltip">
 			<span class="cds--popover-content cds--definition-tooltip">
 				<ng-container *ngIf="!isTemplate(description)">{{description}}</ng-container>
-				<ng-template *ngIf="isTemplate(description)" [ngTemplateOutlet]="description"></ng-template>
+				<ng-template *ngIf="isTemplate(description)" [ngTemplateOutlet]="description" [ngTemplateOutletContext]="{ data }"></ng-template>
```
Comment (Akshat55) [inline src/tooltip/definition-tooptip.component.ts:44]: What are your thoughts on this implicit?

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/tooltip/definition-tooptip.component.ts (hunk 3 above)
Comment (Licen-it) [reply src/tooltip/definition-tooptip.component.ts:44]: I was following probably the old pattern, but it would make sense to have it `$implicit`

**Review by Akshat55 — APPROVED** [review]

**Pull request merged**